src/averageAges.js

```diff
@@ -15,11 +15,23 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  let sumOfMenAges;
+  let averageManAge;
+  const menDiedInGivenCentury = men.filter(person => (
+    Math.ceil(person.died / 100) === century
+  ));
+
+  function getAverageAge(mans) {
+    sumOfMenAges = mans.reduce((acc, person) => {
+      return acc + (person.died - person.born);
+    }, 0);
+    averageManAge = sumOfMenAges / mans.length;
+
+    return averageManAge;
+  }
+
+  return !century ? getAverageAge(men) : getAverageAge(menDiedInGivenCentury);
```

Comment:
1) ternaries should be multilined. It is much easier to read
```suggestion
  return !century 
    ? getAverageAge(men) 
    : getAverageAge(menDiedInGivenCentury);
```
 
 
2) you should start checking boolean values from the positive. Starting with negative is more difficult to read for the human brain 
```suggestion
  return century 
    ? getAverageAge(menDiedInGivenCentury)
    :  getAverageAge(men) ;
```

3) let's try to make this return statement work

```suggestion
  return getAverageAge(men)
```